src/rust/ensogl/lib/core/src/animation/physics/inertia.rs

```diff
@@ -723,3 +727,28 @@ impl Default for EndStatus {
         Self::Normal
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn animation_to_nan() {
+        let mut data = SimulationData::<f32>::new();
+        data.set_value(0.0);
+        data.set_target_value(f32::NAN);
+        data.step(1.0);
+        assert!(data.value().is_nan());
+        assert!(!data.active);
+    }
+
+    #[test]
+    fn animation_from_nan() {
+        let mut data = SimulationData::<f32>::new();
+        data.set_value(f32::NAN);
+        data.set_target_value(0.0);
+        data.step(1.0);
+        assert_eq!(data.value(),0.0);
+        assert!(!data.active);
```

Comment:
please add comments what these tests are for and how they should behave. They are testing a corner cases and this needs additional explanation